testutil/isstrictnilchecker.go

```diff
@@ -0,0 +1,71 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package testutil
+
+import (
+	"fmt"
+	"reflect"
+
+	"gopkg.in/check.v1"
+)
+
+// IsInterfaceNil checks that the value is a nil interface value.
+var IsInterfaceNil = &interfaceNilChecker{
+	&check.CheckerInfo{Name: "IsInterfaceNil", Params: []string{"value"}},
+}
+
+type interfaceNilChecker struct {
+	*check.CheckerInfo
+}
+
+func (*interfaceNilChecker) Check(params []interface{}, names []string) (result bool, errMsg string) {
+	if reflect.ValueOf(params[0]).IsValid() {
+		return false, fmt.Sprintf("expected untyped nil but got %T type", params[0])
+	}
+
+	return true, ""
+}
+
+// NotInterfaceNil checks that the variable is typed and non-nil valued.
+var NotInterfaceNil = &notInterfaceNilChecker{
```

Comment:
isn't this the same as NotNil behaviorally?

Reply:
You're right, my mistake